client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/readclient/FlinkShuffleClientImpl.java

```diff
@@ -533,23 +533,21 @@ private <R> R retrySendMessage(ThrowingExceptionSupplier<R, Exception> supplier)
   }
 
   private boolean shouldRetry(Throwable e) {
-    boolean isIOException =
-        e instanceof IOException
-            || e instanceof TimeoutException
-            || (e.getCause() != null && e.getCause() instanceof TimeoutException)
-            || (e.getCause() != null && e.getCause() instanceof IOException)
-            || (e instanceof RuntimeException
-                && e.getMessage() != null
-                && e.getMessage().startsWith(IOException.class.getName()));
-    return isIOException;
+    return e instanceof IOException
+        || e instanceof TimeoutException
+        || (e.getCause() != null && e.getCause() instanceof TimeoutException)
+        || (e.getCause() != null && e.getCause() instanceof IOException)
+        || (e instanceof RuntimeException
+            && e.getMessage() != null
+            && e.getMessage().startsWith(IOException.class.getName()));
   }
 
   @Override
   public void cleanup(int shuffleId, int mapId, int attemptId) {
     final String mapKey = Utils.makeMapKey(shuffleId, mapId, attemptId);
     super.cleanup(shuffleId, mapId, attemptId);
     if (currentClient != null) {
-      currentClient.remove(mapKey);
+      JavaUtils.closeQuietly(currentClient.remove(mapKey));
```

Comment:
We used connection pools to boost performance. Closing a TransportClient is incorrect because a client might be using a shared connection inside.